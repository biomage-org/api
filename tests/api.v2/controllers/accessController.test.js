// @ts-nocheck
const getUserRoles = require('../../../src/api.v2/helpers/access/getUserRoles');
const userAccessController = require('../../../src/api.v2/controllers/accessController');

jest.mock('../../../src/api.v2/helpers/access/getUserRoles');

const mockRes = {
  json: jest.fn(),
};

const mockUsersList = [
  {
    name: 'Mock Admin',
    email: 'admin@example.com',
    role: 'admin',
  },
  {
    name: 'Mock User',
    email: 'user@example.com',
    role: 'owner',
  },
];

describe('accessController', () => {
  beforeEach(async () => {
    jest.clearAllMocks();
  });

  it('getExperimentUsers works correctly', async () => {
    const mockReq = { params: { experimentId: 'mockExperimentId' } };

    getUserRoles.mockImplementationOnce(
      () => Promise.resolve(mockUsersList),
    );

    await userAccessController.getExperimentUsers(mockReq, mockRes);

    expect(getUserRoles).toHaveBeenCalledWith('mockExperimentId');
    expect(mockRes.json).toHaveBeenCalledWith(mockUsersList);
  });
});
