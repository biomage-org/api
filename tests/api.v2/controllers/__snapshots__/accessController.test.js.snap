// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accessController inviteUser works correctly 1`] = `
Array [
  "mockExperimentId",
  "user@example.com",
  "admin",
  Object {
    "email": "owner@example.com",
  },
]
`;

exports[`accessController revokeAccess works correctly 1`] = `
Array [
  "mockExperimentId",
  "user@example.com",
]
`;
